src/main.rs

```diff
@@ -1,5 +1,23 @@
-#![deny(overflowing_literals, unsafe_code)]
-#![allow(clippy::identity_op)]
+#![deny(
+    overflowing_literals,
+    unsafe_code,
+    rust_2018_idioms,
+    rust_2018_compatibility,
+    nonstandard_style,
+    future_incompatible,
+    missing_copy_implementations,
+    missing_debug_implementations
+)]
+#![warn(clippy::all, clippy::pedantic)]
```

Comment:
I'm strongly opposed to adding the `all` and `pedantic` warnings here. First, even with this PR `cargo clippy` still shows a lot of warnings. Second, as Clippy adds more warnings, I would have to review them and resolve the warnings.

I'm happy that Clippy is not showing any warnings now (and thank you for that!), but any extra warnings should be brought in explicitly one by one.

Reply:
I will revert it.